locale/Makefile

```diff
@@ -4,7 +4,7 @@
 # Run make help to list available targets
 #
 DOMAIN = hammer-cli
```

Comment:
I thought about extracting this from the gemspec too. For example, you could make some logic in Hammer to print the domain for a given gem. If it doesn't match the gem name, it could use the gem metadata.

For reference, we have [this](https://github.com/theforeman/foreman/blob/acafe1aae2eef439a73334635046dfb79be55894/script/plugin_webpack_directories.rb#L44) in Foreman. We could introduce a metadata key like i18n_domain.

Then the shared gemspec becomes 2 lines: gem name and `MAIN_MAKEFILE`.

@ofedoren any thoughts on this?

Reply:
I guess we could go further:
 - Move `VERSION` and `DOMAIN` into `Makefile.def` since it will be shared and used in plugins. No need to redefine those each time. That way, the only thing a plugin will need is a `Makefile` with two lines: importing shared makefile and including it.
 - Define `VERSION` as per your suggestion.
 - Define `metadata` as per your suggestion in gemspec like
 ```ruby
 require 'hammer_cli/i18n'
 # ...
 s.metadata      = { 'in_domain' => HammerCLI::I18n::LocaleDomain.new.domain_name }
 ```
  - Define `DOMAIN` as 
 ```Makefile
 DOMAIN = $(shell ruby -rrubygems -e '
spec = Gem::Specification::load(Dir.glob("../*.gemspec").first);
in_domain = spec.respond_to?(:metadata) && spec.metadata["in_domain"]
puts in_domain || spec.name')
 ```
Pros: we have automated domain definition and only one source of truth. Cons: we need to update all plugins to be consistent since if a plugin won't have 'metadata' entry then the only way we can guess is to use gem's name, which may differ from the domain.

Reply:
Sounds good, the only difference then will be something like `DOMAIN = $(DOMAIN) || $(shell ...)`.